opcua_dbus.c

```diff
@@ -192,9 +233,182 @@ bool dbus_temp_unpack_signal(GVariant *parameters, uint32_t *subscription_id, do
     return true;
 }
 
-void dbus_connect_g_signal(GCallback func)
+void dbus_connect_temp_g_signal(GCallback func)
 {
-    assert(NULL != dbusproxy);
+    assert(NULL != dbusproxy_temp);
     assert(NULL != func);
-    g_signal_connect(dbusproxy, "g-signal", func, NULL);
+    g_signal_connect(dbusproxy_temp, "g-signal", func, NULL);
+}
+
+void dbus_connect_ports_g_signal(GCallback func)
+{
+    assert(NULL != dbusproxy_ports);
+    assert(NULL != func);
+    g_signal_connect(dbusproxy_ports, "g-signal", func, NULL);
+}
+
+/**
+ * get_number_of_ioports:
+ * @brief Get number of I/O ports from dbus.
+ * @return TRUE if successful.
+ */
+bool dbus_get_number_of_ioports(uint32_t *inputs, uint32_t *outputs)
+{
+  assert(NULL != inputs);
+  assert(NULL != outputs);
+  assert(NULL != dbusproxy_ports);
+  gboolean ret = FALSE;
```

Comment:
```suggestion
  bool ret = false;
```

For consistency: `gboolean` is not used in this file but `bool`. I would like to to stick with one of them and not mix.